cmd/zc_enumerator.go

```diff
@@ -222,6 +232,12 @@ func initResourceTraverser(source string, location common.Location, ctx *context
 			return nil, err
 		}
 
+		if sourceURL.Scheme == "http" {
+			httpsRecommendationOnce.Do(func() {
+				glcm.Info(httpsRecommendedNotice)
```

Comment:
A general comment: I'm noticing that all the changes in this PR reference variables named "source", never dest.  However:
(a) for uploads we should encourage HTTPS on the destination.
(b) for S2S we should (and I think already do) encourage HTTPS on the destination (because the source SAS tokens are passed to the destination - and so should not be passed over a plaintext channel).

I think (b) is already addressed somewhere in the code. With a hard error rather than a warning. That's good. Because there's just no way we want people passing SAS's over HTTP.  So (b) is done and its outside the scope of this PR. I just mention it here for completeness..  But I think (a) might be not done yet - unless I'm misreading this PR and it actually does cover destinations.

Does (a) need to be added to this PR?

Reply:
So, initResourceTraverser is run on both the source and the destination.

I'll change the name of source to resource instead, since it does seem to be a bit confusing.